File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if (nums.size() <= 1) return nums.size();

        int max_len = 1;
        unordered_set<int> st;

        for (int i = 0; i < nums.size(); i ++) st.insert(nums[i]);

        for (int num: st) {
            if (st.find(num+1) != st.end()) {
                int curr_len = 0;
                int x = num;
                while (true) {
                    if (st.find(x) == st.end()) break;
                    curr_len ++;
                    x ++;
                }
                max_len = max(curr_len, max_len);
            }
            
           
        }
        return max_len;
    }
};

File: Data Structures & Algorithms/rotate-array/submission-1.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        def rev(nums, st, en):
            while st < en:
                nums[st], nums[en] = nums[en], nums[st]
                st += 1
                en -= 1
        k = k%len(nums)
        rev(nums, 0, len(nums) - k - 1)
        rev(nums, len(nums) - k, len(nums) - 1)
        rev(nums, 0, len(nums)-1)
        